app/src/org/commcare/android/nfc/NfcActivity.java

```diff
@@ -0,0 +1,150 @@
+package org.commcare.android.nfc;
+
+import android.app.Activity;
+import android.app.PendingIntent;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.nfc.NfcAdapter;
+import android.nfc.Tag;
+import android.os.Build;
+import android.os.Bundle;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.commcare.android.logging.ReportingUtils;
+import org.commcare.dalvik.R;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Parent activity that provides all of the functionality common to any NFC action that CommCare
+ * supports
+ *
+ * @author Aliza Stone
+ */
+public abstract class NfcActivity extends Activity {
+
+    protected static final String NFC_PAYLOAD_MULT_TYPES_ARG = "types";
+    protected static final String NFC_PAYLOAD_SINGLE_TYPE_ARG = "type";
+    protected static final String NFC_DOMAIN_ARG = "domain";
+
+    protected NfcManager nfcManager;
+    protected PendingIntent pendingNfcIntent;
+    protected String domainForType;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
```

Comment:
else... return?

Reply:
I'm not sure what you're getting at; what would returning at the end of `onCreate` achieve? `onResume` takes care of finishing the activity if this condition isn't met, since finishing within `onCreate` is incorrect practice.